### DISCUSSION TITLE
How to intercept new registrations and password resets?

### DISCUSSION THREAD

**Discussion by realtebo** [Laravel-Backpack/community-forum#862] [Q&A (Help)]
I am implementing manually, again, the 2 factor auth. 

This time we must handle registration and password recovery, so I need to intercept when a user is succesfully registered to create a 'secret key' into his record 

Also, how to intercept a succesfully password reset? I want to contextually reset also the secret key for 2FA

**Comment by jcastroa87** [comment]
Hello @realtebo, you can follow the controllers on vendor/backpack/crud/src/app/Http/Controllers/Auth to understand the register and recovery password process and events. If you need to customize, there is information about it on our [documentation](https://backpackforlaravel.com/docs/6.x/base-how-to#customizing-the-auth-controllers)

Cheers.

**Comment by realtebo** [comment]
I am following the code, thanks. 

For password reset I find 

event(new PasswordReset($user));

I will try to listen to this event to reset the 2fa

**Comment by jcastroa87** [comment]
I will close the issue due to not activity. Please feel free to re-open or create a new one if needed.

Cheers.
